.travis.yml

```diff
@@ -13,25 +18,40 @@ env:
 
 jobs:
   allow_failures:
-    - rust: nightly
+    - stage: "Build and Test"
+      rust: nightly
   fast_finish: true
   include:
     - name: "Rust: stable - Examples"
+      stage: "Build and Test"
       language: rust
       rust: stable
       services:
         - docker
       install:
-        - (cd examples/truffle; yarn install)
         - docker pull trufflesuite/ganache-cli:v6.7.0
       script:
-        - (cd examples/truffle; yarn run build)
         - docker run -d -p 9545:8545 trufflesuite/ganache-cli:v6.7.0
         - cargo run --example async
         - cargo run --example linked
         - cargo run --package examples-generate
         - cargo run --example rinkeby
 
+    - name: "crates.io"
+      stage: "Deploy"
+      language: rust
+      rust: stable
+      env:
+        - RUSTC_WRAPPER=
+      before_install: skip
+      before_script: skip
+      script: skip
+      deploy:
+        provider: script
+        script: bash ci/deploy.sh
+        on:
+          tags: true
```

Comment:
I'm confused. Why is this required if the stage is only run when a tag is set?

Reply:
Because or travis needs a condition to deploy on, I had it off and got this:
```
Skipping a deployment with the script provider because this branch is not permitted: deploy-test
```
https://travis-ci.org/gnosis/ethcontract-rs/jobs/627650549#L243